tgui/packages/tgui/interfaces/Newscaster.js

```diff
@@ -0,0 +1,487 @@
+              onClick={() => act('togglePhoto')} />
+            {!!security_mode && (
+              <Button
+                icon="ban"
+                content={"D-Notice"}
+                tooltip="Censor the whole channel and it's \
+                  contents as dangerous to the station. Cannot be undone."
+                disabled={!security_mode}
+                onClick={() => act('channelDNotice', {
+                  secure: security_mode,
+                  channel: viewing_channel,
+                })} />
+            )}
+          </Box>
+          <Box>
+            <Button
+              icon="newspaper"
+              content="Print Newspaper"
+              disabled={paper <= 0}
+              onClick={() => act('printNewspaper')} />
+          </Box>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+/** Channel select is the left-hand menu where all the channels are listed. */
+const NewscasterChannelSelector = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    channels = [],
+    viewing_channel,
+  } = data;
+  return (
+    <Section
+      minHeight="100%"
+      width={(window.innerWidth - 410) + "px"}>
+      <Tabs vertical>
+        {channels.map(channel => (
+          <Tabs.Tab
+            key={channel.name}
+            pt={0.75}
+            pb={0.75}
+            mr={1}
+            selected={viewing_channel === channel.ID}
+            icon={channel.censored ? 'ban' : null}
+            textColor={channel.censored ? "Red" : "white"}
+            onClick={() => act('setChannel', {
+              channel: channel.ID,
+            })}>
+            {channel.name}
+          </Tabs.Tab>
+        ))}
+        <Tabs.Tab
+          pt={0.75}
+          pb={0.75}
+          mr={1}
+          textColor="white"
+          color="Green"
+          onClick={() => act('startCreateChannel')}>
+          Create Channel [+]
+        </Tabs.Tab>
+      </Tabs>
+    </Section>
+  );
+};
+
+const processedText = value => {
+  const textHtml = {
+    __html: sanitizeText(marked(value, {
+      breaks: true,
+      smartypants: true,
+      smartLists: true,
+      baseUrl: 'thisshouldbreakhttp',
+    })),
+  };
+  return textHtml;
+};
+
+
+/** This is where the channels comments get spangled out (tm) */
+const NewscasterChannelMessages = (_, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    messages = [],
+    viewing_channel,
+    security_mode,
+    channelCensored,
+    user,
+  } = data;
+  if (channelCensored) {
+    return (
+      <Section color="Red">
+        <b>ATTENTION:</b> Comments cannot be read at this time.<br />
+        Thank you for your understanding, and have a secure day.
+      </Section>
+    );
+  }
+  return (
+    <Section fill scrollable>
+      {messages.map(message => {
+        if (message.channel_num !== viewing_channel) {
+          return;
+        }
+        return (
+          <Section
+            key={message.body}
```

Comment:
What if there's two messages with the same body?

You should use the index provided by `map` instead.